### PR TITLE
Mix precision for tensorflow backend

### PR DESCRIPTION
**Pull request by vbvg2008**
V100, Resnet50 ImageNet benchmark:
batchsize: 128/gpu for full precision, 256/gpu for mixed precision
* single gpu:
	* mixed precision:  627.2 img/sec
	* full precision: 366.1 img/sec
* multi-gpu(4):
	* mixed precision: 1904.64 img/sec
	* full precision: 1203.2 img/sec

### CODE REVIEW

File: fastestimator/network.py [hunk 1]
```diff
@@ -650,6 +652,7 @@ def build(model_fn: Callable[[], Union[Model, List[Model]]],
             automatically generated and assigned.
         weights_path: Path(s) from which to load model weights. If not None, then the number of weight paths provided
             should match the number of models generated by the `model_fn`.
+        mixed_precision: Whether to enable mix precision network operations, only apply to tensorflow models.
```
Comment (TortoiseHam) [inline fastestimator/network.py:655]: only applies to

File: fastestimator/network.py [hunk 2]
```diff
@@ -623,7 +624,8 @@ def _subsample_data(self, data: T, n: int) -> T:
 def build(model_fn: Callable[[], Union[Model, List[Model]]],
           optimizer_fn: Union[str, Scheduler, Callable, List[str], List[Callable], List[Scheduler], None],
           weights_path: Union[str, None, List[Union[str, None]]] = None,
-          model_name: Union[str, List[str], None] = None) -> Union[Model, List[Model]]:
+          model_name: Union[str, List[str], None] = None,
+          mixed_precision: bool = False) -> Union[Model, List[Model]]:
```
Comment (TortoiseHam) [inline fastestimator/network.py:628]: we need a more global way of enforcing mixed precision

Comment (vbvg2008) [reply fastestimator/network.py:628]: people might not want mixed precision on global level

File: fastestimator/network.py (hunk 1 above)
Comment (vbvg2008) [reply fastestimator/network.py]: done

File: fastestimator/backend/update_model.py [hunk 3]
```diff
@@ -59,11 +60,18 @@ def update_model(model: Union[tf.keras.Model, torch.nn.Module],
     """
     loss = reduce_mean(loss)
     if isinstance(model, tf.keras.Model):
+        #  scale up loss for mixed precision training to avoid underflow
```
Comment (TortoiseHam) [inline fastestimator/backend/update_model.py:63]: 2 spaces between # and text

Comment (vbvg2008) [reply fastestimator/backend/update_model.py:63]: nice! done

File: fastestimator/network.py [hunk 4]
```diff
@@ -661,6 +664,11 @@ def _generate_model_names(num_names):
 
     if not hasattr(build, "count"):
         build.count = 0
+    # mix-precision handling
+    if mixed_precision:
+        mixed_precision_tf.set_policy(mixed_precision_tf.Policy('mixed_float16'))
+    else:
+        mixed_precision_tf.set_policy(mixed_precision_tf.Policy('float32'))
```
Comment (TortoiseHam) [inline fastestimator/network.py:671]: if the second call to fe.build() sets the policy back to 'float32' will it still used mixed precision on the earlier models?

Comment (vbvg2008) [reply fastestimator/network.py:671]: yep

**Pull request merged**